cpp/src/cugraph.cu

```diff
@@ -343,7 +445,8 @@ gdf_error gdf_pagerank_impl (gdf_graph *graph,
   ALLOC_MANAGED_TRY((void**)&d_val, sizeof(WT) * nnz , stream);
   ALLOC_MANAGED_TRY((void**)&d_pr,    sizeof(WT) * m, stream);
 
-  cugraph::HT_matrix_csc_coo(m, nnz, (int*)graph->transposedAdjList->offsets->data, (int*)graph->transposedAdjList->indices->data, d_val, d_leaf_vector);
+  //  The templating for HT_matrix_csc_coo assumes that m, nnz and data are all the same type
+  cugraph::HT_matrix_csc_coo(m, nnz, (int *)graph->transposedAdjList->offsets->data, (int *)graph->transposedAdjList->indices->data, d_val, d_leaf_vector);
```

Comment:
If I am not mistaken, typename T is added to specify the type for graph->tranposedAdjList->offsets->data, so should we cast to (T*)?

Reply:
Only spacing difference in my change.  I had started down the path of adding typename T to functions when I discovered that most of the algorithms only work on int32_t anyway.  This particular function does not have a typename T associated with it.

Reply:
And c type casting is dangerous and generally not advised in C++. static_cast<T*>(ptr_to_cast) is much safer in general.

Reply:
As I didn't actually change this code... should I be fixing those things that were already wrong, or should we create a different issue for this?

Reply:
It's up to you, I am not going to block this PR for this, and I guess this isn't the only place for this, it will be a huge amount of work to remove all C type casting in cugraph. What I recommend is to remove C type casting one by one when you see those.

Reply:
I'd suggest that we make this a priority going forward.  I'm happy to try stomping them out as the opportunity arises.  Not really comfortable making those changes at the end of a PR.